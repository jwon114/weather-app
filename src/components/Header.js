import React from 'react';
import PropTypes from 'prop-types';

const Header = (props) => (
  <div className="header">
    <span className="header__location">{props.location}</span>
    <div className="header__time-container">
      <div className="header__time">
        <span className="header__dot brown"></span>
        <span className="header__dot tan"></span>
        <span className="header__dot gold"></span>
        <span className="header__time_value">{props.time}</span>
        <span className="header__dot brown"></span>
        <span className="header__dot tan"></span>
        <span className="header__dot gold"></span>
      </div>
    </div>
    <span className="header__temperature">{props.temperature}&deg;</span>
  </div>
);

Header.propTypes = {
  location: PropTypes.string,
  time: PropTypes.string,
  temperature: PropTypes.number
}

export default Header;