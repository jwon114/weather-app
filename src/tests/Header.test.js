import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { render } from '@testing-library/react';
import Header from '../components/Header';

test('should render Header correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('renders location, time and temperature', () => {
  const { getByText } = render(<Header 
    location={'london'} 
    time={'10:24 GMT'}
    temperature={'24'} />)

  const locationElement = getByText(/london/i);
  const timeElement = getByText('10:24 GMT');
  const temperatureElement = getByText('24');

  expect(locationElement).toBeInTheDocument();
  expect(timeElement).toBeInTheDocument();
  expect(temperatureElement).toBeInTheDocument();
});