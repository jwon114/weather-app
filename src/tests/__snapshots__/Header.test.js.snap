// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="header"
>
  <span
    className="header__location"
  />
  <div
    className="header__time-container"
  >
    <div
      className="header__time"
    >
      <span
        className="header__dot brown"
      />
      <span
        className="header__dot tan"
      />
      <span
        className="header__dot gold"
      />
      <span
        className="header__time_value"
      />
      <span
        className="header__dot brown"
      />
      <span
        className="header__dot tan"
      />
      <span
        className="header__dot gold"
      />
    </div>
  </div>
  <span
    className="header__temperature"
  >
    Â°
  </span>
</div>
`;
